pkg/controllers/work/patch_util.go

```diff
@@ -105,8 +106,22 @@ func setModifiedConfigurationAnnotation(obj runtime.Object) error {
 		return err
 	}
 	// set the last applied annotation back
-	annotations[lastAppliedConfigAnnotation] = string(modified)
-	return metadataAccessor.SetAnnotations(obj, annotations)
+	annotations[LastAppliedConfigAnnotation] = string(modified)
+	if err = metadataAccessor.SetAnnotations(obj, annotations); err != nil {
+		return err
+	}
+	annotations, err = metadataAccessor.Annotations(obj)
+	if err != nil {
+		return err
+	}
+	if err := validation.ValidateAnnotationsSize(annotations); err != nil {
+		klog.InfoS(fmt.Sprintf("setting last applied config annotation to empty, %s", err))
+		annotations[LastAppliedConfigAnnotation] = ""
+		if err = metadataAccessor.SetAnnotations(obj, annotations); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
```suggestion
	annotations[LastAppliedConfigAnnotation] = string(modified)
	if err := validation.ValidateAnnotationsSize(annotations); err != nil {
		klog.V(2).InfoS(fmt.Sprintf("setting last applied config annotation to empty, %s", err))
		annotations[LastAppliedConfigAnnotation] = ""
	}
	return metadataAccessor.SetAnnotations(obj, annotations)
```
you don't need to set & get the annotation multiple times.

Please update the func comment to indicate the "set the annotation to empty" logic.